packages/manager/apps/telecom/src/app/telecom/pack/slots/hostedEmail/detail/detail.controller.js

```diff
@@ -0,0 +1,237 @@
+import assignIn from 'lodash/assignIn';
+import filter from 'lodash/filter';
+import isObject from 'lodash/isObject';
+import map from 'lodash/map';
+import snakeCase from 'lodash/snakeCase';
+
+export default class PackHostedEmailDetailCtrl {
+  /* @ngInject */
+  constructor(
+    $q,
+    $state,
+    $http,
+    $translate,
+    TucToast,
+    OvhApiPackXdsl,
+    OvhApiPackXdslHostedEmail,
+  ) {
+    this.$q = $q;
+    this.$state = $state;
+    this.$http = $http;
+    this.$translate = $translate;
+    this.TucToast = TucToast;
+    this.OvhApiPackXdsl = OvhApiPackXdsl;
+    this.OvhApiPackXdslHostedEmail = OvhApiPackXdslHostedEmail;
+  }
+
+  $onInit() {
+    this.askForDelete = false;
+    this.accountDeleting = false;
+    this.loading = true;
+
+    this.passwordRules = [
+      {
+        id: 'length',
+        caption: this.$translate.instant(
+          'hosted_email_detail_change_password_rule_size',
+        ),
+        validator(str) {
+          return str && str.length > 7 && str.length < 21;
+        },
+      },
+      {
+        id: 'specialChar',
+        caption: this.$translate.instant(
+          'hosted_email_detail_change_password_rule_special',
+          { list: '#{}()[]-|@=*+/!:;' },
+        ),
+        validator: /^[\w~"#'{}(\\)[\]\-|\\^@=*+/!:;.,?<>%*µÀÁÂÃÄÅàáâãäåÒÓÔÕÖØòóôõöøÈÉÊËèéêëÇçÌÍÎÏìíîïÙÚÛÜùúûüÿÑñ]+$/,
+        immediateWarning: true,
+      },
+    ];
```

Comment:
Imo we shouldn't have to handle password validation rules. 
Could the regexp be extracted to constants ?